staging/src/k8s.io/apiserver/pkg/registry/generic/rest/proxy.go

```diff
@@ -178,6 +190,17 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 		}
 	}
 
+	if rawResponseCode != http.StatusSwitchingProtocols {
+		// If the backend did not upgrade the request, finish echoing the response from the backend to the client and return, closing the connection.
+		glog.Infof("Proxy upgrade error, status code %d", rawResponseCode)
+		_, err := io.Copy(requestHijackedConn, backendConn)
+		if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
nit: Not sure why the original committer added this check. I would just log the err if not nil instead of matching against a specific string.